XPNet.CLR/Scenery/XPlaneScenery.cs

```diff
@@ -82,23 +83,64 @@ public IXPInstance CreateInstance(IEnumerable<string> inDataRefs)
 			return new XPInstance(instanceRef);
 		}
 
-		public void Draw(int lighting, int earthRelative, XPLMDrawInfo_t[] drawInfos)
+		public void Draw(int lighting, int earthRelative, XPDrawInfo[] drawInfos)
 		{
-			fixed (XPLMDrawInfo_t* p = drawInfos)
+			fixed (XPDrawInfo* p = drawInfos)
 			{
 				PluginBridge.Log.Log($"Drawing {drawInfos.Length} objects");
 				PluginBridge.ApiFunctions.XPLMDrawObjects(m_objectRef, drawInfos.Length, p, lighting, earthRelative);
 			}
 		}
 	}
 
+	/// <summary>
+	/// This is the draw info struct that needs to be passed to 
+	/// <see cref="XPLMDrawObjects"/> and corresponds to XPLMDrawInfo_t
+	/// in the X-Plane Plugin API. Note that this structure is exposed directly
+	/// for performance reasons.
+	/// </summary>
+	[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack = 2)]
+	public struct XPDrawInfo
+	{
+		//Note that the structSize needs to be also an instance member
+		//of the struct such that it is layed out correcly for X-Plane
+		//Therefore, the size of the struct is initially written into a
+		//static member in the static constructor and copied into the
+		//instance member in the instance constructor.
+		private static readonly int m_structSizeInit;
+		internal readonly int m_structSize;
+		public float x;
+		public float y;
+		public float z;
+		public float pitch;
+		public float heading;
+		public float roll;
```

Comment:
We should name all these with UpperCamelCase since they're now part of the XPNet API.  (It will work even though the names are lower-case in the C version, b/c the mapping doesn't care about the names, only the positions).  So like X, Y, Z, Pitch, Heading, Roll.

Reply:
Done in 9cf6c1da